### PR TITLE
automate the ironbank docker context generation

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

Automate the docker context generation for the IronBank releases, this will allow us to move away from creating those docker context manually in a different repository, since the Platform Release team already provide the automation.

## Implementation details

A similar approach was done for [Kibana](https://github.com/elastic/kibana/pull/89933) and [Elasticsearch](https://github.com/elastic/elasticsearch/pull/64336)

I don't know if other teams are actually building those docker, but for now this is the very first iteration to move away from the manual steps required.

## DoD build context content

As required the build context contains the following files:
- `Dockerfile` with specific requirements for DoD (registry args, no internet dependencies, healthcheck, ...)
- `hardening_manifest.yaml` with all required dependencies
- `LICENSE` specific for DoD
- `README.md` with specific content for DoD
- `config/` (optional) directory containing all config files to include into the Docker image
- `scripts/` directory containing all scripts to include into the Docker image (example: entrypoint...)

## Tasks

- [x] Exclude if not running for `x86_64`.
- [ ] Enable configuration in the DRA so it can be automated.
- [ ] Gather feedback from the teams.

## Test

if you `mage ironbank` then you will see the `.tar.gz` in the `build/distributions` folder, for instance:

```bash
$ mage ironbank
>> prepareIronbankBuild: prepare the IronBank container context.
>> saveIronbank: save the IronBank container context.
>> creating TAR file from directory: ./, target: ../../build/distributions/elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz
$ ls -ltra build/distributions 
total 16
drwxr-x--- 4 vmartinez staff  128 Jul  7 09:20 .
drwxr-x--- 4 vmartinez staff  128 Jul 19 12:03 ..
-rw-r--r-- 1 vmartinez staff 9476 Jul 19 12:03 elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz
-rw-r--r-- 1 vmartinez staff  186 Jul 19 12:03 elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz.sha512
$ ls -ltra build/             
total 0
drwxr-x---  4 vmartinez staff  128 Jul  7 09:20 distributions
drwxr-xr-x 46 vmartinez staff 1472 Jul 19 12:03 ..
drwxr-x---  4 vmartinez staff  128 Jul 19 12:03 .
drwxr-xr-x  7 vmartinez staff  224 Jul 19 12:03 elastic-agent-ironbank-8.4.0-docker-build-context
$ cd build/distributions 
$ tar -xvf elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz 
x ./
x Dockerfile
x LICENSE
x README.md
x config
x config/docker-entrypoint
x hardening_manifest.yaml
$ ls -l
total 48
-rw-r--r-- 1 vmartinez staff  3928 Jul 19 12:03 Dockerfile
-rw-r--r-- 1 vmartinez staff 16642 Jul 19 12:03 LICENSE
-rw-r--r-- 1 vmartinez staff  1781 Jul 19 12:03 README.md
drwxr-xr-x 3 vmartinez staff    96 Jul 19 12:03 config
-rw-r--r-- 1 vmartinez staff  9476 Jul 19 12:03 elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz
-rw-r--r-- 1 vmartinez staff   186 Jul 19 12:03 elastic-agent-ironbank-8.4.0-docker-build-context.tar.gz.sha512
-rw-r--r-- 1 vmartinez staff  2567 Jul 19 12:03 hardening_manifest.yaml
```

### CI

`ironbank` is now enabled in the package stage and artifacts are generated:

```bash
$ gsutil cp gs://fleet-ci-artifacts/elastic-agent/pull-requests/pr-679/elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz .    
Copying gs://fleet-ci-artifacts/elastic-agent/pull-requests/pr-679/elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz...
/ [1 files][  9.2 KiB/  9.2 KiB]                                                
Operation completed over 1 objects/9.2 KiB.                                      
➜  tar xvf elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz 
x ./
x Dockerfile
x LICENSE
x README.md
x config
x config/docker-entrypoint
x hardening_manifest.yaml

```

## Issue

Similar to https://github.com/elastic/apm-server/pull/8537

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by amannocci — APPROVED** [review]

File: dev-tools/packaging/templates/ironbank/README.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+**NOTE**
+
+This functionality is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+---
+
+### Installation instructions
+
+Please follow the documentation on [Quick start](https://www.elastic.co/guide/en/ingest-management/7.17/ingest-management-getting-started.html).
```
Comment (v1v) [inline dev-tools/packaging/templates/ironbank/README.md.tmpl:18]:
```suggestion
Please follow the documentation on [Quick start](https://www.elastic.co/guide/en/fleet/{{ .MajorMinor }}/fleet-elastic-agent-quick-start.html).
```

File: dev-tools/packaging/templates/ironbank/README.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Where to file issues and PRs
+
+- [Issues](https://github.com/elastic/beats/issues)
+- [PRs](https://github.com/elastic/beats/pulls)
```
Comment (v1v) [inline dev-tools/packaging/templates/ironbank/README.md.tmpl:23]:
```suggestion
- [Issues](https://github.com/elastic/elastic-agent/issues)
- [PRs](https://github.com/elastic/elastic-agent/pulls)
```

File: dev-tools/packaging/templates/ironbank/Dockerfile.tmpl [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  && rm  ${ELASTIC_PRODUCT}-${ELASTIC_STACK}-${OS_AND_ARCH}.tar.gz
+
+# Support arbitrary user ids
+# Ensure that group permissions are the same as user permissions.
+# This will help when relying on GID-0 to run Kibana, rather than UID-1000.
+# OpenShift does this, for example.
+# REF: https://docs.okd.io/latest/openshift_images/create-images.html
+RUN chmod -R g=u /usr/share/${ELASTIC_PRODUCT}
+
+# Create auxiliar folders and assing default permissions.
```
Comment (v1v) [inline dev-tools/packaging/templates/ironbank/Dockerfile.tmpl:28]:
```suggestion
# Create auxiliary folders and assigning default permissions.
```

**Comment by v1v** [comment]
### Test

```
➜  elastic-agent gsutil cp gs://fleet-ci-artifacts/elastic-agent/pull-requests/pr-679/elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz .
Copying gs://fleet-ci-artifacts/elastic-agent/pull-requests/pr-679/elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz...
/ [1 files][  9.3 KiB/  9.3 KiB]                                                
Operation completed over 1 objects/9.3 KiB.                                      
➜  elastic-agent tar xvf elastic-agent-ironbank-8.4.0-SNAPSHOT-docker-build-context.tar.gz 
x ./
x Dockerfile
x LICENSE
x README.md
x config
x config/docker-entrypoint
x hardening_manifest.yaml
➜  elastic-agent grep 8.4 *                                                                                                      
Dockerfile:ARG ELASTIC_STACK=8.4.0
README.md:Please follow the documentation on [Quick start](https://www.elastic.co/guide/en/fleet/8.4/fleet-elastic-agent-quick-start.html).
README.md:License](https://github.com/elastic/beats/blob/8.4/licenses/ELASTIC-LICENSE.txt),
README.md:[here](https://www.elastic.co/guide/en/beats/libbeat/current/release-notes-8.4.0.html).
grep: config: Is a directory
hardening_manifest.yaml:- "8.4.0"
hardening_manifest.yaml:  ELASTIC_STACK: "8.4.0"
hardening_manifest.yaml:  org.opencontainers.image.version: "8.4.0"
hardening_manifest.yaml:  - filename: "elastic-agent-8.4.0-linux-x86_64.tar.gz"
hardening_manifest.yaml:    url: "<artifact_path>/elastic-agent-8.4.0-linux-x86_64.tar.gz"
```

**Pull request merged**